ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOModel.scala

```diff
@@ -211,11 +217,38 @@ class H2OMOJOModel(override val uid: String) extends H2OMOJOModelBase[H2OMOJOMod
 
 object H2OMOJOModel extends H2OMOJOReadable[PyH2OMOJOModel] with H2OMOJOLoader[PyH2OMOJOModel] {
 
+  private def removeMetaField(json: JsonElement): JsonElement = {
+    if (json.isJsonObject) {
+      json.getAsJsonObject.remove("__meta")
+      json.getAsJsonObject.entrySet().asScala.map(_.getValue).foreach(removeMetaField)
```

Comment:
Can we merge `map` and `foreach` logic together? If we did that, we would avoid materialization of the collection produced by `map`.